File: Excel/579 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/579 Rotated Strings_2.xlsx"
input = read_excel(path, range = "A1:B10")
test  = read_excel(path, range = "C2:D6", col_names = c("Var1", "Var2")) %>%
  arrange(Var1)

is_rotated = function(string1, string2) {
  is_0_rot = string1 == string2
  is_rot = str_detect(paste0(string1, string1), string2)
  is_length_equal = nchar(string1) == nchar(string2)
  return(is_rot & !is_0_rot & is_length_equal)
}

result = expand.grid(input$String1, input$String2) %>%
  mutate(Var1 = as.character(Var1),
         Var2 = as.character(Var2)) %>%
  mutate(is_rotated = map2_lgl(Var1, Var2, is_rotated)) %>%
  filter(is_rotated) %>%
  select(-is_rotated) %>%
  arrange(Var1)

all.equal(result, test, check.attributes = FALSE)
#> [1] TRUE

File: Excel/500-599/579/579 Challenge.py
import pandas as pd
import numpy as np

path = "579 Rotated Strings_2.xlsx"

input = pd.read_excel(path, usecols="A:B", nrows = 10)
test = pd.read_excel(path, usecols="C:D", skiprows=1, nrows=5, header=None, names=["Var1", "Var2"]).sort_values(by="Var1").reset_index(drop=True)

def is_rotated(original, check):
    return check in original + original and original != check and len(original) == len(check)

result = pd.DataFrame(np.array(np.meshgrid(input["String1"], input["String2"])).T.reshape(-1, 2), columns=["Var1", "Var2"])
result["IsRotated"] = result.apply(lambda x: is_rotated(x["Var1"], x["Var2"]), axis=1)
result = result[result["IsRotated"]].sort_values(by="Var1").reset_index(drop=True).drop(columns="IsRotated")

print(result.equals(test))  # True